main.go

```diff
@@ -15,13 +16,29 @@ import (
 )
 
 func main() {
+
+	var err error
+
+	input := parseFlags()
+	if input.IsBenchmark {
+		err = benchmark.Run(input)
+	} else {
+		_, err = worker.Run(input)
+	}
+
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func parseFlags() models.Input {
 	// run options
 	runTimeout := flag.Duration("run", 30*time.Second, "stop run after this duration")
 	flushTimeout := flag.Duration("flush", 10*time.Second, "wait timeout for agent flush")
 	seed := flag.Int64("seed", time.Now().Unix(), "random seed")
 
 	// convenience for https://www.elastic.co/guide/en/apm/agent/go/current/configuration.html
-	serviceName := flag.String("service-name", "", "service name") // ELASTIC_APM_SERVICE_NAME
+	serviceName := flag.String("service-name", "hey-service", "service name") // ELASTIC_APM_SERVICE_NAME
```

Comment:
I'd prefer this uses the value in `ELASTIC_APM_SERVICE_NAME` if set, and fall back to `hey-service`

Reply:
like so c85098b258b1b4648abd53ecb22c3c0dd32b668e?